### PR TITLE
implement related:card on-hover animation

### PR DESCRIPTION
**Pull request by nickhsine**
### Animation Demo
https://kanbanflow.com/task-attachment/XkYlkn1ROVTA

### KanbanFlow Card
https://kanbanflow.com/t/LDJmYY7c

### CODE REVIEW

File: packages/react-article-components/src/components/related/card.js [hunk 1]
```diff
@@ -110,133 +145,140 @@ const Title = styled.h3`
...
-  display: inline-block;
-  font-size: 14px;
-  font-weight: ${typography.font.weight.bold};
-  color: #808080;
-  line-height: 1;
-  padding: 5px 10px;
-  &::before {
-    content: '更多內容';
+  componentWillUnmount() {
+    this._blockRef = undefined
```
Comment (YuCJ) [inline packages/react-article-components/src/components/related/card.js]: React will clear the `ref.current` on `componentWillUnmount`
https://reactjs.org/docs/refs-and-the-dom.html#adding-a-ref-to-a-dom-element

File: packages/react-article-components/src/components/related/card.js [hunk 2]
```diff
@@ -110,133 +145,140 @@ const Title = styled.h3`
...
-`
 
-const PublishedDate = styled.span`
-  font-size: 12px;
-  line-height: 2;
-  color: #afafaf;
+  render() {
+    const { category, title, desc, date, thumbnail } = this.props
+    const localeDate = date
+      ? new Date(date).toLocaleString('zh-hant', {
```
Comment (YuCJ) [inline packages/react-article-components/src/components/related/card.js]: `zh-Hant` or `zh-Hant-TW`.

And the params may not be supported by Android webview and IE under 11.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleString#Browser_compatibility

**Comment by nickhsine** [comment]
updated.

File: packages/react-article-components/src/components/related/index.js [hunk 3]
```diff
@@ -88,74 +95,96 @@ const Descriptor = styled.div`
...
   }
 
   static defaultProps = {
     data: [],
   }
 
+  constructor(props) {
+    super(props)
+
+    this.lastWindowWidth = 0
```
Comment (YuCJ) [inline packages/react-article-components/src/components/related/index.js]: Move `lastWindowWidth` into `this.state`?

File: packages/react-article-components/src/components/related/list.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+                  <Card {...item} />
+                </components.Link>
+              </Item>
+            )
+          }}
+        </DynamicComponentsContext.Consumer>
+      )
+    })
+
+    if (cards.length === 0) {
```
Comment (YuCJ) [inline packages/react-article-components/src/components/related/list.js]: We can use `if (data.length === 0) { return null }` and move this before mapping data

File: packages/react-article-components/src/components/related/list.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    const cards = _.map(data, item => {
+      return (
+        <DynamicComponentsContext.Consumer key={item.id}>
+          {components => {
+            return (
+              // Use `stretch` to make sure every Item(Card)
+              // having the same height while mounting.
+              // Then, use `flex-start` instead to make each Item(Card)
+              // show description with different height while hovering.
+              <Item alignSelf={!isMounted ? 'stretch' : 'flex-start'}>
```
Comment (YuCJ) [inline packages/react-article-components/src/components/related/list.js]: Nice :+1:

**Comment by nickhsine** [comment]
updated.

**Pull request merged**